list-navigation/lambda/list-provider.ts

```diff
@@ -4,7 +4,7 @@
 
 // representation of a page token; simply a raw string as the real representation
 // will differ with each list provider instance
-export type PageToken = string;
```

Comment:
why did the type of the page token need to change? this may not serlize to the String type defined in ACDL correctly if it's not a string

Reply:
PageToken in case of DDBListProvider if of the type `Record<string, AttributeValue>`. So I made this change, as different providers may have greatly differing PageToken type.